pkg/server/environment_create.go

```diff
@@ -127,10 +136,38 @@ func (s *Server) environmentCreate(c *gin.Context) {
 			},
 		},
 	}
+	if len(repoInfo.URL) > 0 {
+		logrus.Debugf("clone code from %s", repoInfo.URL)
+		repoName := repoInfo.URL[strings.LastIndex(repoInfo.URL, "/")+1:]
+		expectedPod.Spec.InitContainers = append(expectedPod.Spec.InitContainers, v1.Container{
+			Name:  "git-cloner",
+			Image: "alpine/git",
+			Args:  []string{"clone", "--", repoInfo.URL, "/code"},
+			VolumeMounts: []v1.VolumeMount{
+				{
+					Name:      "code-dir",
+					MountPath: "/code",
+				},
+			},
+		})
+		expectedPod.Spec.Containers[0].VolumeMounts = append(expectedPod.Spec.Containers[0].VolumeMounts, v1.VolumeMount{
+			Name:      "code-dir",
+			MountPath: fmt.Sprintf("/home/envd/%s", repoName),
+		})
+		expectedPod.Spec.Volumes = append(expectedPod.Spec.Volumes, v1.Volume{
+			Name: "code-dir",
+			VolumeSource: v1.VolumeSource{
+				HostPath: &v1.HostPathVolumeSource{
+					Path: fmt.Sprintf("/var/envd/code/%s", req.Name),
```

Comment:
Where is the mount point in envd's main container? `/var/envd/code/<containername>`?

Reply:
I think it should be `/home/envd/<project>`

Reply:
This is the host path. Currently code dir is mounted with the repo name (at L141).

Reply:
Could we mount it to ${ENVD_WORKDIR}? We may need to add a new image `label ai.tensorchord.envd.name`. Then use `/home/envd/<name>`.

Because the sshd server is running at the dir, thus when you run ssh, the process's cwd is also there.

```
            "Env": [
                "SHELL=/usr/bin/zsh",
                "USER=envd",
                "ENVD_WORKDIR=/home/envd/golang",
                "PATH=/opt/conda/envs/envd/bin:/opt/conda/bin:/home/envd/.local/bin:/usr/local/j
ulia/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
                "LC_ALL=en_US.UTF-8"
            ],
```

Reply:
sgtm

Reply:
fixed